src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -164,9 +170,8 @@ public void addSubExpressionGraph(ExpressionGraph subGraph,
     // map of subgraph edge -> new edge to be inserted
     Map<ExpressionEdge, ExpressionEdge> exprEdgeMap = new HashMap<>();
     subGraph.getEdges().forEach(e -> {
-        // copy source/target for now since ExpressionEdge doesn't like creating with null on both
```

Comment:
(Related to comment about getSource being able to return null)
In lines 167, nothing seems to be verifying that mainGraphInput.getSource() isn't null or mainGraphOutput.getTarget() isn't null. Also, they're really far from the only place they're used(line 206).
```
ExpressionVertex inputVertex = mainGraphInput.getSource();
ExpressionVertex outputVertex = mainGraphOutput.getTarget();
```